apiserver/cmd/main.go

```diff
@@ -30,11 +31,20 @@ var (
 	rpcPortFlag        = flag.String("rpcPortFlag", ":8887", "RPC Port")
 	httpPortFlag       = flag.String("httpPortFlag", ":8888", "Http Proxy Port")
 	collectMetricsFlag = flag.Bool("collectMetricsFlag", true, "Whether to collect Prometheus metrics in API server.")
+	logFile            = flag.String("logFilePath", "", "sync log to local file")
 )
 
 func main() {
 	flag.Parse()
 
+	if *logFile != "" {
+		flagSet := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+		klog.InitFlags(flagSet)
```

Comment:
I am not familiar with the `flag` package. Does the order of `klog.InitFlags(flagSet)` and `flagSet.Set(...)` matter? i.e. 

```go
_ = flagSet.Set("alsologtostderr", "true")
_ = flagSet.Set("logtostderr", "false")
_ = flagSet.Set("log_file", *logFile)
klog.InitFlags(flagSet)
```

Reply:
InitFlags make the flagSet link to the inner config of klog.
then use flagSet.Set can direct change the config of klog.
If we switch the order, it can not work

Reply:
Thanks!